### PR TITLE
fix: [CDS-72164]: fixed the logic of enabling save btn on form changes

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

after fix: 
https://github.com/harness/harness-core-ui/assets/57372986/add7456f-dac4-4c6b-9b6f-47fa269df31d

### CODE REVIEW

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 1]
```diff
@@ -341,7 +341,26 @@ export function FormikInputSetForm(props: FormikInputSetFormProps): React.ReactE
           }}
           formName="inputSetForm"
           validationSchema={NameIdSchema}
-          validate={validateValues}
+          validate={values => {
+            const isIpSetPipelineChanged = isEqual(values?.pipeline, inputSet?.pipeline)
+            const fieldsOutOfPipelineFrmvalues = {
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:346]: There is typo here, I guess

**Review by chetanvekariya-harness — COMMENTED** [review]
We always enabled save button inside InputSetForm component. Don't we need any change there? You might have to temporary changes that I did to fix QA release blocker issue.

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 2]
```diff
@@ -362,22 +381,14 @@ export function FormikInputSetForm(props: FormikInputSetFormProps): React.ReactE
         >
           {formikProps => {
             formikRef.current = formikProps
-            const handleChange = (): void => {
-              // form dirty is determined on formvalues.pipeline
-              // This is used to determine to show the unsaved changes button
-              const isIpSetPipelineChanged = isEqual(formikRef?.current?.values?.pipeline, inputSet?.pipeline)
-              // Save is enabled on the whole form
-              const isIpSetFormDirty = isEqual(formikRef?.current?.values, inputSet)
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:370]: @sahithikolichala Input set form has a child form gitSyncForm, with this line we were capturing if anything in child form changes but with new code we do not see anything to handle same. So, please test the changes for both old and new GitExperience.  cc: @chetanvekariya-harness

Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:370]: This repo can be used to save pipeline and inputsets - https://github.com/wings-software/git_sync_chetan

Comment (sahithikolichala) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:370]: this is a goodcatch but inputSet does not seem to have the branch and other details
@sunnykesh-harness  @chetanvekariya-harness

**Review by chetanvekariya-harness — COMMENTED** [review]
Tried checking this fix in my local using below input set link
[Open Input Set](https://localhost:8181/ng/account/px7xd_BFRCi-pfWPYXVjvw/cd/orgs/default1/projects/Chetan_Non_Git_Sync/pipelines/AWS_SAM_Pipeline_Edit_1906/input-sets/AWS_SAM_Pipeline_IS_1?storeType=INLINE)
I am getting unsaved changes as soon as input set loads.

**Comment by kalyansai-harness** [comment]
@sahithikolichala please update the PR with right description with problem statement and the solution (how you implemented and the approach you took)

**Pull request closed without merging**